### DISCUSSION TITLE
what happen if database got updated before Prisma schema.

### DISCUSSION THREAD

**Discussion by aboveyunhai** [prisma/prisma#23619] [Q&A] [answered]
### Question

Forget about the migration, 
we all know that if the schema contains non-existing field in DB table, it will raise error. 
But I wonder how Prisma handles the opposite situation, such a case that if the schema is in sync with the current database table, and then the table added an optional (`nullable`) column.
Without updating the current schema, will Prisma still run correctly (as the missing field from schema will be automatically filled by `null`)? or it will raise error due to inconsistency, or something else like data corruption? just wonder the default behavior for schema inconsistency and resilience. 

### How to reproduce (optional)
Not applicable

### Expected behavior (optional)
Not applicable

### Information about Prisma Schema, Client Queries and Environment (optional)
Not applicable

**Comment by ludralph** [comment, marked as answer]
Hi @aboveyunhai 👋 

If a new optional (nullable) column is added to the database table without updating the Prisma schema, Prisma will not raise an error. This is because Prisma does not enforce the schema on the database.

However, it's important to note that the new column will not be recognized by Prisma until you update your Prisma schema and run the introspection command. This means that you won't be able to interact with the new column through Prisma until you do so.

If you want to ensure that your Prisma schema is always in sync with your database, you should regularly run the introspection command. This command will update your Prisma schema to match the current state of your database.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
